src/lib/y2storage/guided_proposal.rb

```diff
@@ -205,17 +206,32 @@ def candidate_devices_with_empty_partition_table(devicegraph)
       devices.select { |d| d.partition_table && d.partitions.empty? }
     end
 
-    # Copy of the original settings including some calculated and necessary
-    # values (mainly candidate_devices), in case they were not present
+    # Returns the target sizes to make the proposal
+    #
+    # @return [Array<Symbol>]
+    def target_sizes
+      [:desired, :min]
+    end
+
+    # Settings used by each attempt of proposal
     #
     # @return [ProposalSettings]
     def populated_settings
-      return @populated_settings if @populated_settings
+      @populated_settings ||= settings.dup
+    end
 
-      populated = settings.dup
-      populated.candidate_devices ||= disk_analyzer.candidate_disks.map(&:name)
+    # List of candidate devices
+    #
+    # If they are not given in the settings, disks found by the analyzer will be used
+    #
+    # TODO: improve this documentation
```

Comment:
Ey @joseivanlopez 

Could you please help me improving the documentation of this method?

Thanks in advance :)

Reply:
Sure, although I am not specially good in that area :)

```
# Candidate devices grouped for different proposal attempts
#
# When some candidate devices are indicated in the settings, the proposal is tried with all that candidate devices.
# But when no candidate devices are given, different attempts are done by using different sets of candidate devices. First, each available device is used lonely to make the proposal, and if no proposal was possible with any individual disk, a last attempt is done by using all available devices as candidate disks.
#
# @example
#
#   settings.candidate_devices #=> ["/dev/sda", "/dev/sdb"]
#   settings.group_candidate_devices #=> [["/dev/sda", "/dev/sdb"]]
#
#   settings.candidate_devices #=> nil
#   settings.group_candidate_devices #=> [["/dev/sda"], ["/dev/sdb"], ["/dev/sda", "/dev/sdb"]]